### PR TITLE
[ios][image-manipulator] support base64 data URIs

### PR DESCRIPTION
**Pull request by mnightingale**
# Why

* Contrary to the documentation `web` and `android` already support base64 data URIs so I figured I may as well add it to `iOS`.

# How

* Added `iOS` support for reading data URIs
* Updated the documentation
* Added loading from data URI test
* Spotted that `apps/test-suite/TestModules.js` wasn't including the image manipulator!
  * There are a bunch of failing tests for most platforms; I'm making seperate PR's to fix them and other issues rather than complicating this PR.

# Test Plan

* Added Image Manipulator to `apps/test-suite/TestModules.js`
* Added a base64 data URI test
* Checked in bare-expo app

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: apps/test-suite/tests/ImageManipulator.js [hunk 1]
```diff
@@ -24,6 +24,18 @@ export async function test(t) {
...
+      t.it('returns valid image from base64 data URL', async () => {
+        // 1x1 red image
+        const result = await ImageManipulator.manipulateAsync(
+          'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z8DwHwAFBQIAX8jx0gAAAABJRU5ErkJggg==',
+          [{ resize: { width: 100, height: 100 } }]
+        );
+        t.expect(result).toBeDefined();
+        t.expect(typeof result.uri).toBe('string');
+        t.expect(typeof result.width).toBe('number');
+        t.expect(typeof result.height).toBe('number');
```
Comment (bbarthec) [inline apps/test-suite/tests/ImageManipulator.js:36]: Can't we be more specific about the results here? 🤔 
```suggestion
        t.expect(result.width).toBe(100);
        t.expect(result.height).toBe(100);
```

File: packages/expo-image-manipulator/ios/EXImageManipulator/EXImageManipulatorModule.m [hunk 2]
```diff
@@ -47,6 +47,16 @@ - (void)setModuleRegistry:(UMModuleRegistry *)moduleRegistry
   }
   NSString *path = [url.path stringByStandardizingPath];
 
+  if ([[url scheme] isEqualToString:@"data"]) {
+    NSData *data = [[NSData alloc]initWithContentsOfURL:url];
```
Comment (bbarthec) [inline packages/expo-image-manipulator/ios/EXImageManipulator/EXImageManipulatorModule.m:51]:
```suggestion
    NSData *data = [[NSData alloc] initWithContentsOfURL:url];
```

**Review by bbarthec — COMMENTED** [review]
Looks very promising! Thank you for this one! 👏

File: apps/test-suite/tests/ImageManipulator.js (hunk 1 above)
Comment (mnightingale) [reply apps/test-suite/tests/ImageManipulator.js]: I copied what the test above it does, I've changed it but I think the intention of both was more can it read and process the file (from local file or base64) not whether the resize action is doing what is requested.

**Review by bbarthec — APPROVED** [review]
Great job on this one 👍 👏

**Pull request merged**